ec/src/models/bn/g1.rs

```diff
@@ -23,14 +22,32 @@ impl<P: BnParameters> From<G1Affine<P>> for G1Prepared<P> {
     }
 }
 
+impl<P: BnParameters> From<G1Projective<P>> for G1Prepared<P> {
```

Comment:
this pattern of `impl<> From<> for G1Prepared<>` repeats in each model - could we maybe have a macro to generate this impls?

Reply:
Good points. And it seems that only BN, BLS, BW also implements `is_zero`, but not MNT46. Any reason why?

Reply:
It depends entirely on what the corresponding pairing implementations require; the MNT46 impls don't seem to require that method.